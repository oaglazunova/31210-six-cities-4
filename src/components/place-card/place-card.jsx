import React from 'react';
import PropTypes from 'prop-types';

import {TOfferPropTypes} from '../app/app.model.js';

export const PlaceCard = (props) => {
  const {data, handleTitleClick, handleCardHover} = props;

  const handleMouseOverCard = () => {
    handleCardHover(data.id);
  };

  const handleMouseLeaveCard = () => {
    handleCardHover(null);
  };

  const onTitleClick = () => {
    handleTitleClick(data.id);
  };

  return (
    <article className="cities__place-card place-card" onMouseEnter={handleMouseOverCard} onMouseLeave={handleMouseLeaveCard}>
      {data.mark &&
        <div className="place-card__mark"><span>{data.mark}</span></div>
      }
      <div className="cities__image-wrapper place-card__image-wrapper">
        <a href="#"><img className="place-card__image" src={data.imgSrc ? data.imgSrc : `img/no-places.png`} width="260" height="200" alt={data.description} /></a>
      </div>

      <div className="place-card__info">
        <div className="place-card__price-wrapper">

          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{data.priceValue}</b>
            {data.priceText && <span className="place-card__price-text">&nbsp;{data.priceText}</span>}
          </div>

          <button className="place-card__bookmark-button button" type="button">
            <svg className="place-card__bookmark-icon" width="18" height="19"><use xlinkHref="#icon-bookmark" /></svg>
            <span className="visually-hidden">To bookmarks</span>
          </button>
        </div>

        <div className="place-card__rating rating">
          <div className="place-card__stars rating__stars">
            <span style={{'width': `100%`}} />
            <span className="visually-hidden">Rating</span>
          </div>
        </div>

        <h2 className="place-card__name"><a onClick={onTitleClick} href="#">{data.title}</a></h2>

        <p className="place-card__type">{data.type}</p>
      </div>
    </article>
  );
};

PlaceCard.propTypes = {
  data: TOfferPropTypes,
  handleTitleClick: PropTypes.func,
  handleCardHover: PropTypes.func,
};
