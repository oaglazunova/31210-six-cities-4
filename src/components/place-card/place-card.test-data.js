export default {
  id: 501,
  title: `Characteristic apartment on the third floor with sunny balcony`,
  isPremium: false,
  imgSrc: `img/apartment-03.jpg`,
  gallery: [
    `img/apartment-03.jpg`,
    `img/studio-photos.jpg`,
    `img/amsterdam.jpg`,
  ],
  priceValue: 600,
  priceText: ``,
  rating: null,
  description: [
    `Note: No agency fees applicable when renting this apartment directly through Housing Agency.`,
    `Characteristic apartment on the third floor with sunny balcony (south) in the beautiful and quiet Burmanstraat nearby the popular area of the Amstel river. Shops, restaurants, and Pijp within walking distance. Metro line close by.`],
  type: `house`,
  bedrooms: null,
  maxGuests: 4,
  utensils: [
    `Wi-Fi`,
    `Kitchen`,
    `Dishwasher`,
    `Cable TV`,
    `Fridge`,
  ],
  host: {
    name: `Maria`,
    avatar: `img/avatar-angelina.jpg`,
    isPro: false,
  },
  coordinates: [52.3909553943508, 4.85309666406198],
};
