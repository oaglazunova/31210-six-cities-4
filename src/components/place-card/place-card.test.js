import React from 'react';
import renderer from 'react-test-renderer';
import {PlaceCard} from './place-card.jsx';

import testData from './place-card.test-data.js';

describe(`Place Card content rendered`, () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(
        <PlaceCard data={testData} handleTitleClick={()=>{}} handleCardHover={()=>{}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
