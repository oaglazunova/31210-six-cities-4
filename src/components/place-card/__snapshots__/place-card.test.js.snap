// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Place Card content rendered renders correctly 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt={
          Array [
            "Note: No agency fees applicable when renting this apartment directly through Housing Agency.",
            "Characteristic apartment on the third floor with sunny balcony (south) in the beautiful and quiet Burmanstraat nearby the popular area of the Amstel river. Shops, restaurants, and Pijp within walking distance. Metro line close by.",
          ]
        }
        className="place-card__image"
        height="200"
        src="img/apartment-03.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          â‚¬
          600
        </b>
        
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Characteristic apartment on the third floor with sunny balcony
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      house
    </p>
  </div>
</article>
`;
